packages/typegpu/src/core/slot/derived.ts

```diff
@@ -60,6 +66,12 @@ function createBoundDerived<T>(
   innerDerived: TgpuDerived<T>,
   pairs: SlotValuePair[],
 ): TgpuDerived<T> {
+  if (getResolutionCtx()) {
+    throw new Error(
+      'Cannot create tgpu.derived objects at the resolution stage.',
+    );
+  }
+
```

Comment:
Binding derived should be possible during resolution, only defining new derived should be outlawed 🔫